### PR TITLE
ENH: Add dtype of categories to repr of CategoricalDtype

### LINKED ISSUE #52179: ENH: CategoricalDtype __repr__ should show dtype of categories
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Right now the repr only shows the categories and if they are ordered, but since we can back Categoricals with arrow or nullable dtypes or generally different NumPy dtypes (e.g. int32) we should also show the dtype of the categories:

```
ser1 = Series(pd.Categorical(Index([1, 2, 3], dtype="int32")))
ser2 = Series(pd.Categorical(Index([1, 2, 3], dtype="int64")))

tm.assert_series_equal(ser1, ser2)
```

This shows
```
Attribute "dtype" are different
[left]:  CategoricalDtype(categories=[1, 2, 3], ordered=False)
[right]: CategoricalDtype(categories=[1, 2, 3], ordered=False)
```

which is not very helpful.

### Feature Description

Add something like ``category_dtype="int32"`` to the repr.

### Alternative Solutions

Not sure what you could do apart from checking explicitly for the categories dtype yourself.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #52179 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
```
=================================== FAILURES ===================================
__ [doctest] pandas.core.dtypes.dtypes.CategoricalDtype._from_values_or_dtype __
246             If ``CategoricalDtype``, cannot be used together with
247             `categories` or `ordered`.
248 
249         Returns
250         -------
251         CategoricalDtype
252 
253         Examples
254         --------
255         >>> pd.CategoricalDtype._from_values_or_dtype()
Expected:
    CategoricalDtype(categories=None, ordered=None)
Got:
    CategoricalDtype(categories=None, ordered=None, categories_dtype=None)

/home/runner/work/pandas/pandas/pandas/core/dtypes/dtypes.py:255: DocTestFailure
___ [doctest] pandas.io.json._table_schema.convert_json_field_to_pandas_type ___
167     ------
168     ValueError
169         If the type of the provided field is unknown or currently unsupported
170 
171     Examples
172     --------
173     >>> convert_json_field_to_pandas_type({"name": "an_int", "type": "integer"})
174     'int64'
175 
176     >>> convert_json_field_to_pandas_type(
Expected:
    CategoricalDtype(categories=['a', 'b', 'c'], ordered=True)
Got:
    CategoricalDtype(categories=['a', 'b', 'c'], ordered=True, categories_dtype=object)
```

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -429,13 +429,19 @@ def __eq__(self, other: Any) -> bool:
     def __repr__(self) -> str_type:
         if self.categories is None:
             data = "None"
+            dtype = "None"
```
Comment (mroeschke) [inline pandas/core/dtypes/dtypes.py:432]: Do we have a test that hits this branch?

File: doc/source/whatsnew/v2.1.0.rst [hunk 2]
```diff
@@ -37,6 +37,7 @@ Other enhancements
 - Improve error message when having incompatible columns using :meth:`DataFrame.merge` (:issue:`51861`)
 - Improved error message when creating a DataFrame with empty data (0 rows), no index and an incorrect number of columns. (:issue:`52084`)
 - :meth:`arrays.SparseArray.map` now supports ``na_action`` (:issue:`52096`).
+- Add dtype of categories to display information of :class:`CategoricalDtype` (:issue:`52179`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.1.0.rst:40]:
```suggestion
- Add dtype of categories to ``repr`` information of :class:`CategoricalDtype` (:issue:`52179`)
```

File: pandas/core/dtypes/dtypes.py (hunk 1 above)
Comment (phofl) [reply pandas/core/dtypes/dtypes.py:432]: Yes, can't recall which one exactly but got a failure before I added this

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**